plugin/web.py

```diff
@@ -47,27 +46,38 @@ def advance(self, recvSize=1024):
             return False
 
         rlist, wlist = select.select([self.sock], [self.sock], [], 0)[:2]
+        self.sock.settimeout(1.0)
 
         if rlist:
-            msg = self.sock.recv(recvSize)
-            if not msg:
-                self.close()
-                return False
+            while True:
+                try:
+                    msg = self.sock.recv(recvSize)
+                except: #ConnectionResetError or socket.timeout
```

Comment:
Please explicitly catch these exceptions and let other ones go through. Don't want to do a catch-all here and doing so is stronger than having a comment.

Reply:
Done!